cpp/src/arrow/util/parallel.h

```diff
@@ -44,5 +44,21 @@ Status ParallelFor(int num_tasks, FUNCTION&& func) {
   return st;
 }
 
+// A parallelizer that takes a `Status(int)` function and calls it with
+// arguments between 0 and `num_tasks - 1`, in sequence or in parallel,
+// depending on the input boolean.
+
+template <class FUNCTION>
+Status OptionalParallelFor(bool use_threads, int num_tasks, FUNCTION&& func) {
+  if (use_threads) {
+    return ParallelFor(num_tasks, func);
```

Comment:
`std::forward<FUNCTION>(func)` can be more efficient.

Reply:
Awesome! 
Thank you for the good suggestion.